### DISCUSSION TITLE
How to construct a content part from other parts?

### DISCUSSION THREAD

**Discussion by lbcsy** [OrchardCMS/OrchardCore#13766] [Q&A]
Hi,

Just played with Orchard Core a little bit and I am most interested in the Content Management area, where you can create content type, content part etc. But I haven't figure out how to add another Content Part into a content part while you are creating it. For example, in a healthcare app, people's name has been used at a lot of places, so I would like to create it as a content part so that I can attach it to content type like Patient, Doctor, Nurse etc. In this name part (HumanName), on top of the fields like first name, last name etc., I would also like to attach another part to it to represent the name's effective period, this part (Period) contains two date fields: Start and End. The reason I like to define Period as part because it looks quite common and can be resued in many other places too, alternative, you will have to add Start and End fields all the time when you need something to represent a period.

There maybe some reasons that Orchard has already considered and decided not to support adding part to a part (interestingly, it seems it can achieve adding content type to content part via a field), can anyone shed some light on the reasons?

**Comment by hishamco** [comment]
Parts are attached to content types, for example, `TitlePart` could be attached to many content types while it's commonly used

**Reply by lbcsy** [reply]
Is Orchard's Content modeling mainly from html page perspective? i.e., to help organize/map contents to a page, not much to do with data modeling itself.

**Comment by ns8482e** [comment]
@lbcsy in orchard's concept, 
- You can attach a content field to your content part. 
- You can attach content part to your content item. 
- You can also add content field directly to your content item.

In your case you can build a custom field eg called Duration that will have StartDate and EndDate. And you can attach that Duration field to your content Part

**Reply by lbcsy** [reply]
Yes, this can be one option. It soon becomes not the chosen one if you need to do some calculation on StartDate and EndDate such as disable the content item if it is not within the Duration.

**Reply by ns8482e** [reply]
You are free to implement your own business logic behind the data model you define using orchard core's content management

**Reply by bcampardou** [reply]
What about using a ListPart in a ContentType that let you add other content items ?

**Reply by ns8482e** [reply]
Yes

**Reply by lbcsy** [reply]
The disadvantage of using a ListPart is that parts in the list is kind of anonymous, instead of, I want to express that in this particular part, it has a Period part.

**Reply by jtkech** [reply]
Yes, we don't support nested parts but we support nested types by using `BagPart`s allowing to have embedded items of given content type(s), knowing that any content type can have parts but also has a default part (same name as the type) allowing to add fields directly on it.
